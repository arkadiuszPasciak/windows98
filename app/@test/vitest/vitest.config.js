import { defineProject, mergeConfig } from "vitest/config"
import sharedConfig from "@windows98/configs/vitest/shared"
import tsconfigPaths from "vite-tsconfig-paths"

export default mergeConfig(
	sharedConfig,
	defineProject({
		plugins: [tsconfigPaths({ loose: true })],
	})
)
