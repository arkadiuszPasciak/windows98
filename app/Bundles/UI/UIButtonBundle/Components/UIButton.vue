<template>
  <button
    class="UIButton"
    :class="`size-${size}`"
    type="button"
    :disabled="disabled"
  >
    <slot name="icon-left" />
    <slot />
    <slot name="icon-right" />
  </button>
</template>

<script setup lang="ts">
import type { PropType } from "vue"

defineProps({
	size: {
		type: String as PropType<
			"small" | "medium" | "media-player" | "tic-tac-toe"
		>,
		default: "medium",
	},
	disabled: {
		type: Boolean,
		default: false,
	},
})
</script>

<style lang="scss" scoped src="@APP|Bundles/UIButtonBundle/Styles/UIButton.styles.scss" />
