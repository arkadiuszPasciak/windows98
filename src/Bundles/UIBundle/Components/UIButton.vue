<template>
  <button class="UIButton" :class="`v-${variant} s-${size}`" type="button">
    <slot name="icon-left"></slot>
    <slot></slot>
    <slot name="icon-right"></slot>
  </button>
</template>

<script setup lang="ts">
  import { PropType } from 'vue'

  defineProps({
    variant: {
      type: String as PropType<'primary'>,
      required: true,
    },
    size: {
      type: String as PropType<'small' | 'medium'>,
      required: true,
    },
  })
</script>

<style lang="scss" scoped>
  @import '@Bundles/UIBundle/Styles/UIButton.styles.scss';
</style>
