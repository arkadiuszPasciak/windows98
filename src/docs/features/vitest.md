# [Vitest](https://vitest.dev/)

## Description

Vitest is a modern and fast testing framework specifically designed for Vite-powered projects. Developed by the [Vite team](https://vitejs.dev/team.html), Vitest leverages Vite's architecture to provide a seamless testing experience with features like native ESM support, rapid test execution, and isolated testing environments. It offers compatibility with various front-end libraries and frameworks, such as Vue, React, and Svelte. With its straightforward configuration and integration with Vite, Vitest aims to enhance the developer experience by providing an efficient and enjoyable testing workflow for Vite-based web applications.

## Commands

[Available commands in the application](/reference/cli.html#vitest)
