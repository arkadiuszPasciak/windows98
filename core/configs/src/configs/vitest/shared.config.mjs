import { defineConfig } from 'vitest/config'

export default defineConfig({
    test: {
        environment: "jsdom",
        exclude: [
            "**/build/**",
            "**/dist/**",
            "**/node_modules/**",
            "**/@bundler/**",
            "**/@linter/**",
            "**/@test/**",
        ],
        include: [
            "**/*.unit.ts"
        ],
        globals: true,
    },
})
