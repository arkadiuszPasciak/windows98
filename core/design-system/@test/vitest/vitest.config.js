import { defineProject, mergeConfig } from "vitest/config"
import sharedConfig from "@windows98/configs/vitest/shared"

export default mergeConfig(
	sharedConfig,
	defineProject({/* here custom configs */ })
)
