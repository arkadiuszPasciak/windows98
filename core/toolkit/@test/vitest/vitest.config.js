import sharedConfig from "@windows98/configs/vitest/shared"
import { defineProject, mergeConfig } from "vitest/config"

export default mergeConfig(
	sharedConfig,
	defineProject({
		/* here custom configs */
	}),
)
