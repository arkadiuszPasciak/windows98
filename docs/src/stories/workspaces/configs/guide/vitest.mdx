import { Meta, Title } from "@storybook/addon-docs/blocks";

<Meta title="Workspaces/Configs/Guide/Vitest" />

<Title>âš¡ Vitest</Title>

Complete guide to using Vitest configuration from `@windows98/configs` for unit and integration testing.

## Overview

Vitest configuration provides fast, modern testing setup optimized for TypeScript and modern JavaScript frameworks within the Windows98 ecosystem.

## Configuration Path

```
@windows98/configs/vitest/shared
```

### Install Dependencies

```bash
# Install vitest
pnpm add -D vitest

# Install the configs package
pnpm add -D @windows98/configs
```

## Basic Usage

### Simple Setup

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig(sharedConfig)
```

### With Custom Settings

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig({
  ...sharedConfig,
  test: {
    ...sharedConfig.test,
    environment: 'jsdom',
    setupFiles: ['./src/test-setup.ts']
  }
})
```

## Package.json Scripts

```json
{
  "scripts": {
	"test:unit:run": "vitest run -c ./@test/vitest/vitest.config.js",
	"test:unit:watch": "vitest -c ./@test/vitest/vitest.config.js",
  }
}
```

