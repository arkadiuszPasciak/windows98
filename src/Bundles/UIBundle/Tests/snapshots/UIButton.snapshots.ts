import { mount } from '@vue/test-utils'
import { expect, it, describe } from 'vitest'
import UIButton from '@Bundles/UIBundle/Components/UIButton.vue'

const sizeSmallPrimaryWrapper = mount(UIButton, {
  props: {
    variant: 'primary',
    size: 'small',
  },
  slots: {
    default: 'Click Me!',
  },
})

const sizeMediumPrimaryWrapper = mount(UIButton, {
  props: {
    variant: 'primary',
    size: 'medium',
  },
  slots: {
    default: "Don't Click Me!",
  },
})

describe('[UIBundle]<Snapshots>(UIButton)', async () => {
  it('renders correctly component: size small and variant primary', () => {
    expect(sizeSmallPrimaryWrapper.element).toMatchSnapshot()
  })

  it('renders correctly component: size medium and variant primary', () => {
    expect(sizeMediumPrimaryWrapper.element).toMatchSnapshot()
  })
})
