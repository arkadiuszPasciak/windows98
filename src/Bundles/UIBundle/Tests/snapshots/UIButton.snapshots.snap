// Vitest Snapshot v1

exports[`[UIBundle]<Snapshots>(UIButton) > renders correctly  component: size small and variant primary component 1`] = `
<button
  class="UIButton variant-primary size-small"
  data-v-cf1bb3a9=""
  type="button"
>
  
  
  
  Click Me!
  
  
  
</button>
`;

exports[`[UIBundle]<Snapshots>(UIButton) > renders correctly component: size medium and variant primary 1`] = `
<button
  class="UIButton variant-primary size-medium"
  data-v-cf1bb3a9=""
  type="button"
>
  
  
  
  Don't Click Me!
  
  
  
</button>
`;

exports[`[UIBundle]<Snapshots>(UIButton) > renders correctly component: size small and variant primary 1`] = `
<button
  class="UIButton variant-primary size-small"
  data-v-cf1bb3a9=""
  type="button"
>
  
  
  
  Click Me!
  
  
  
</button>
`;

exports[`[UIBundle]<Snapshots>(UIButton) > renders correctly size medium and variant primary component 1`] = `
<button
  class="UIButton variant-primary size-medium"
  data-v-cf1bb3a9=""
  type="button"
>
  
  
  
  Don't Click Me!
  
  
  
</button>
`;

exports[`[UIBundle]<Snapshots>(UIButton) > renders correctly size small and variant primary component 1`] = `
<button
  class="UIButton variant-primary size-small"
  data-v-cf1bb3a9=""
  type="button"
>
  
  
  
  Click Me!
  
  
  
</button>
`;
